src/components/SignUp.tsx

```diff
@@ -15,54 +15,195 @@ import {
   PrimaryButtonProps,
 } from '../interfaces/interfaces';
 
-const Title = ({children}: TitleProps) => (
+import LoadingModal from './LoadingModal';
+
+const Title = ({ children }: TitleProps) => (
   <Text style={styles.title}>{children}</Text>
 );
 
-const InputField = ({label, value, onChangeText}: InputProps) => (
+const InputField = ({ label, value, onChangeText, error, secureTextEntry }: InputProps) => (
   <View>
-    <Text style={styles.inputLabel}>{label}</Text>
-    <TextInput value={value} onChangeText={onChangeText} style={styles.input} />
+    <View style={styles.inputLabelContainer}>
+      <Text style={styles.inputLabel}>{label}</Text>
+      {error ? (<Text style={styles.error}>{error}</Text>) : null}
+    </View>
+    <TextInput
+      value={value}
+      onChangeText={onChangeText}
+      style={styles.input}
+      secureTextEntry={secureTextEntry}
+    />
   </View>
 );
 
-const CustomCheckbox = ({value, onValueChange, label}: CheckboxProps) => {
+const CustomCheckbox = ({ value, onValueChange, label, error }: CheckboxProps) => {
   const handleCheckboxChange = () => {
     onValueChange(!value);
   };
 
   return (
-    <View style={styles.checkbox}>
-      <CheckBox
-        value={value}
-        onValueChange={handleCheckboxChange}
-        style={value ? styles.checkboxSelected : styles.checkboxUnselected}
-      />
-      <Text>{label}</Text>
-    </View>
+    <>
+      <View style={styles.checkbox}>
+        <CheckBox
+          value={value}
+          onValueChange={handleCheckboxChange}
+          style={value ? styles.checkboxSelected : styles.checkboxUnselected}
+        />
+        <Text>{label}</Text>
+      </View>
+
+      {error ? (<Text style={styles.errorCheckBox}>{error}</Text>) : null}
+
+    </>
```

Comment:
Is better to a file for each component that you're using on your screen. Same thing for the styles, you can have the styles on another file so your code is not to long. Also, you can create another folder within the source folder named screens and place your screens there such as the signup one, login, etc.

Reply:
Requested changes are done on the latest commit